lya/data.py

```diff
@@ -1,43 +1,73 @@
 from .conf import lya_data_path
-
+import numpy as np
+import matplotlib.pyplot as plt
 
 class Data(object):
-    def __init__(self, x, y, y_unc):
-        self.x = x
-        self.y = y
-        self.y_unc = y_unc
 
+    def __init__(self,
+                 k=None,
+                 P_k=None,
+                 P_k_unc=None,
+                 ):
 
-def load_lya_data(filename=lya_data_path):
-    """Load lyman-alpha data
+        self.k = k
+        self.P_k = P_k
+        self.P_k_unc = P_k_unc
 
-    Parameters
-    ----------
-    filename : str
-        Path to ly-a data
+    def load_lya_txt(self, filename=lya_data_path, return_arrays=False):
+        """Load lyman-alpha data text file
 
-    Returns
-    -------
-    data : Data object
-        K, matter power spectrum, and p(k) uncertainty arrays,
-        encapsulated in Data object
-    """
+        Parameters
+        ----------
+        filename : str
+            Path to ly-a data with 3 columns: k, P_k, P_k_unc
 
+        Returns
+        -------
+        None
 
-def plot_pk(params=None, filename=lya_data_path):
-    """Plot p(k) power spectrum data and theory, with panel for residuals
-    
-    Parameters
-    ----------
-    params : dict or None
-        Cosmological parameter dictionary; if None, then plot only data
+        sets the class attributes k, P_k, P_k_unc
 
+        k is the wave vector [h/Mpc]
+        P_k is the matter power spectrum P(k) [Mpc/h^3]
+        P_k_unc is the P(k) uncertainty [Mpc/h^3]
+        """
 
-    Returns
-    -------
-    fig : matplotlib Figure instance
-    """
+        assert filename.endswith(".txt"), "filename must be a .txt file"
+
+        # read the text file
+        k, P_k, P_k_unc = np.loadtxt(filename, unpack=True)
+
+        if return_arrays:
+            return k, P_k, P_k_unc
+        else:
+            self.k, self.P_k, self.P_k_unc = k, P_k, P_k_unc
+
+    def plot_pk(self, logscale=True,
```

Comment:
Same here-- let's keep this a top-level function (rather than a method) that creates the `Data` object within it and uses that to plot.